lib/types.nix

```diff
@@ -243,6 +243,43 @@ rec {
       merge = mergeOneOption;
     };
 
+    posix = {
```

Comment:
IMHO these `posix` types should be a separate PR. At first glance, they strike me as a bit overengineered and not really necessary to the writers stuff.

Reply:
Alright, I removed them